### PR TITLE
Remove NativeActivity

### PR DESCRIPTION
**Pull request by splhack**

### CODE REVIEW

**Comment by splhack** [comment]
It includes a fix for multithread JNI call problem.
http://www.cocos2d-x.org/forums/6/topics/48163?r=48218#message-48218

File: cocos/2d/platform/android/java/src/org/cocos2dx/lib/Cocos2dxActivity.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+import android.os.Build;
+import android.os.Bundle;
+import android.os.Message;
+import android.view.ViewGroup;
+import android.util.Log;
+import android.widget.FrameLayout;
+
+import java.io.File;
+
+import org.acra.ACRA;
```
Comment (WenhaiLin) [inline cocos/2d/platform/android/java/src/org/cocos2dx/lib/Cocos2dxActivity.java]: What's ACRA?  It's necessary? It cause build fails.

**Comment by splhack** [comment]
@Dhilan007 Thanks! I forgot to remove the line. I've updated the pull request.

**Comment by WenhaiLin** [comment]
@splhack  You miss implementation of 'nativeSetEditTextDialogResult' in  'CCEditBoxImplAndroid.cpp'. Could you re-add it ASAP?Thanks very much.

**Comment by splhack** [comment]
@Dhilan007 Eh, I added 'nativeSetEditTextDialogResult' to 'cocos/2d/platform/android/jni/Java_org_cocos2dx_lib_Cocos2dxHelper.cpp'. Cocos2dxHelper.java has this one.

**Comment by WenhaiLin** [comment]
Thanks

**Pull request merged**